wgsl/index.bs

```diff
@@ -3730,25 +3728,26 @@ as an entry point.
 When configuring the stage in the pipeline, the entry point is specified by providing
 the [SHORTNAME] module and the entry point's function name.
 
-An entry point function must have no parameters,
-and its return type must be [=void=].
+The parameters of an entry point have to be within [=Entry point IO type=]s.
+They have [=storage classes/in=] storage class.
+The return type of an entry point has to be of an [=Entry point IO type=], or [=void=].
 
 <div class='example wgsl global-scope' heading='Entry Point'>
   <xmp>
-    [[builtin(position)]]   var<out> gl_Position  : vec4<f32>;
-    [[builtin(frag_coord)]] var<in> coord_in : vec4<f32>;
-    [[location(0)]] var<out> color : vec4<f32>;
-
     [[stage(vertex)]]
-    fn vtx_main() -> void { gl_Position = vec4<f32>(); }
-       // OpEntryPoint Vertex %vtx_main "vtx_main" %gl_Position
+    fn vtx_main() -> [[builtin(position)]] vec4<f32> {
+      return vec4<f32>(0.0, 0.0, 0.0, 1.0);
+    }
+       // OpEntryPoint Vertex %vtx_main "vtx_main" %(gl_Position)
 
     [[stage(fragment)]]
-    fn frag_main() -> void { color = vec4<f32>(coord_in.x, coord_in.y, 0.0, 1.0); }
-       // OpEntryPoint Fragment %frag_main "frag_main" %color %coord_in
+    fn frag_main([[builtin(frag_coord)]] coord_in: vec4<f32>) -> [[location(0)]] vec4<f32> {
+      return vec4<f32>(coord_in.x, coord_in.y, 0.0, 1.0);
+    }
+       // OpEntryPoint Fragment %frag_main "frag_main" %(gl_FragColor) %coord_in
 
     [[stage(compute)]]
-    fn main() -> void { }
+    fn main() { }
```

Comment:
I believe the `-> void` is still required. A function must have a return type.

Reply:
Well, #1219 was accepted, so it's not really a blocker here. I'll add those back here, anyway.

Reply:
Until it's in the spec, I don't think we should use it.